dlang/plugin-impl/build.gradle.kts

```diff
@@ -1,156 +1,55 @@
+
 fun properties(key: String) = providers.gradleProperty(key).get()
 
 plugins {
     id("org.gradle.idea")
     id("java")
-    alias(libs.plugins.gradleIntelliJPlugin)
+    alias(libs.plugins.gradleIntelliJPlatform)
 }
 
-val javaVersion = properties("javaVersion")
-val ideaVersion = properties("ideaVersion")
-
-val publishChannels = listOf(properties("publishChannels"))
-
-val baseIDE = "idea"
-
 // cobertura.coverageFormats = ["html", "xml"] // coveralls plugin depends on xml format report
 // cobertura.coverageSourceDirs = [sourceSets.main.java.srcDirs, sourceSets.main.kotlin.srcDirs, "gen"]
 // cobertura.coverageEncoding = "UTF-8"
 // cobertura.coverageExcludes = [ ".*uk.co.cwspencer.*" ]
 
-intellij {
-    pluginName.set("intellij-dlanguage")
-    version.set(ideaVersion)
-    //type.set("CL") // maybe helpful to change this when trying other IDE. can be "IC", "IU", "JPS", or "CL"
-    updateSinceUntilBuild.set(false)
```

Comment:
I set plugin name here to control the name of the built artifact, otherwise it was defaulting to _DLanguage_ which it seems to pick up from `settings.gradle.kts`.

Also, setting _updateSinceUntilBuild_ to false and handling manually seems to give better control over supporting a range of IDEA versions. Last time I let the gradle plugin do it the resulting plugin.xml wasn't ideal.

Reply:
I see it put in the generated XML, and the updateSinceUntilBuild option disapeared. Now if you don’t set the values in the pluginConfiguration section, it won’t be set in the resulting XML. But btw, the xml generation was greatly improved and setting config in pluginConfiguration section properly populate the xml attributes.